sfputil/main.py

```diff
@@ -1112,6 +1112,61 @@ def run_firmware(port_name, mode):
 
     return status
 
+def is_fw_switch_done(port_name):
+    """
+        Make sure the run_firmware cmd is done
+        @port_name:
+        Returns 1 on success, and exit_code = -1 on failure
+    """
+    status = 0
+    physical_port = logical_port_to_physical_port_index(port_name)
+    sfp = platform_chassis.get_sfp(physical_port)
+
+    try:
+        api = sfp.get_xcvr_api()
+    except NotImplementedError:
+        click.echo("This functionality is currently not implemented for this platform")
+        sys.exit(ERROR_NOT_IMPLEMENTED)
+
+    try:
+        MAX_WAIT = 600 # 60s timeout.
+        fw_info = api.get_module_fw_info()
+        cnt = 0
+        is_busy = 1 if (fw_info['status'] == False) and (fw_info['result'] is not None) else 0
+        while is_busy and cnt < MAX_WAIT:
+            time.sleep(0.1)
```

Comment:
May not need to check this frequently. It also brings a lot of workload to the module. Checking every 1 or 2 seconds should be OK without causing too much time delay.

Reply:
Agree, already revised.